### ISSUE TITLE
Very high loss

### ISSUE THREAD

**Issue by vaishnavm217** [sfzhang15/SFD#38]
Hi,
I tried converting code from caffe to tensorflow. I didn't do the 'subsampling of fc layers to conv', rest all is same your paper implementation.
The loss which I am getting is very high and diverse, in range of 1 to 2000. So just wanted to ask if you faced any high loss values during training. 
The resultant is giving 0.2 mAP in widerface_val for 12k iterations (Batch size 1 as GPU is pretty small). Will train more to get better results.

**Comment by vaishnavm217** [comment]
One thing I just found.
When I increase the match threshold to 0.5, loss reduced drastically.
So indication of high loss is due to the loss threshold?

**Comment by vaishnavm217** [comment]
One more factor is the lambda. Adding that is giving high loss again.

**Comment by sfzhang15** [comment]
@vaishnavm217 
Did you do the normlization for conv3_3, conv4_3 and conv5_3 layers as caffe-ssd?

**Comment by vaishnavm217** [comment]
@sfzhang15 I had done l2 normalization, but didnt do the constant intialization as mentioned inyour proto.txt
Did that now still loss is ranges of 3000-1000

**Comment by vaishnavm217** [comment]
I think if I didnt implement maxout labels, and reduced lambda, it might become stable. maxout labels is giving me dimension errors do I will omit that.

**Comment by vaishnavm217** [comment]
I think I need to check logic in the scale compensation method. 
For now I eval the bare framework. Should I still keep 0.35 as threshold and lamda as 4? 
This is similar to S<sup>3</sup>FD (F) in your ablative study

**Comment by sfzhang15** [comment]
@vaishnavm217 
1. The l2 normalization in ssd-caffe is across the channel dimention.
2. If your conf loss is normlizated only by the number of positives, then lamda should be 1 (like ssd-caffe).
3. You can temporarily ignore max-out background label and the scale compensation matching strategy.
4. For IoU threshold, it should be 0.35. For lamda, please refer to 1.

**Comment by vaishnavm217** [comment]
Just to see if I am implementing correctly, SFD has only one anchor per feature map that too only at one aspect ratio.
But how will that make face scale continuous?

**Comment by sfzhang15** [comment]
@vaishnavm217 
anchor scale is discrete, since we only set 6 anchor scales. While face scale can be considered continuous, since each scale of faces that we need to detect exist in the training set.

**Comment by vaishnavm217** [comment]
Really sorry I asked the wrong question.
You have used only one type and scale of anchor box in this model. How does this handle the continuous scale of faces in dataset? or this positioning gave the most optimal results??

**Comment by sfzhang15** [comment]
@vaishnavm217 
We use 6 detection layers and each of them is associated with one scale anchor, so there are 6 scales of anchor. As the paper said, the scale of face is continuous, while the scale of anchor is discrete, so some scales of face will not match enough anchors. This is why we propose anchor matching compesation strategy.
